### PR TITLE
Set the cursor in the correct position whenever `CustomDrugEntrySheet` is opened 

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.shortcut.com/simpledotorg/story/5046/custom-drug-entry-sheet-ui-improvements

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt [hunk 1]
```diff
@@ -45,14 +45,21 @@ class CustomDrugEntryUpdate : Update<CustomDrugEntryModel, CustomDrugEntryEvent,
       model: CustomDrugEntryModel,
       drug: Drug
   ): Next<CustomDrugEntryModel, CustomDrugEntryEffect> {
+    val cursorPosition = if (drug.dosage != null) cursorPositionFromDosage(drug.dosage) else 0
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt:48]: Do we need the to return 0 twice here? `cursorPositionFromDosage` is going to return us 0, can we instead check the nullability inside the function and have one condition (either x position or 0 position. WDYT?

File: app/src/test/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdateTest.kt [hunk 2]
```diff
@@ -242,7 +244,32 @@ class CustomDrugEntryUpdateTest {
...
+        .given(drugFrequencyChoiceItemsLoaded)
+        .whenEvent(DrugFetched(drug))
+        .then(
+            assertThatNext(
+                hasModel(drugFrequencyChoiceItemsLoaded
+                    .drugNameLoaded(drug.name)
+                    .dosageEdited(drug.dosage)
+                    .frequencyEdited(drug.frequency)
+                    .rxNormCodeEdited(drug.rxNormCode)
+                    .drugInfoProgressStateLoaded()),
```
Comment (Honey14) [inline app/src/test/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdateTest.kt:272]: Can we extract the model out for readability?

File: app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt [hunk 3]
```diff
@@ -67,6 +67,7 @@ class CustomDrugEntryUpdate : Update<CustomDrugEntryModel, CustomDrugEntryEvent,
       prescription: PrescribedDrug
   ): Next<CustomDrugEntryModel, CustomDrugEntryEffect> {
     val frequency = DrugFrequency.fromMedicineFrequency(prescription.frequency)
+    val cursorPosition = if (prescription.dosage != null) cursorPositionFromDosage(prescription.dosage) else 0
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt:70]: I see why we have 2 conditions now. Apart from having `cursorPositionFromDosage` usage twice? Is there some other reason that I'm missing? 
If not, can we have one condition for these into their individual functions.

**Review by Honey14 — CHANGES_REQUESTED** [review]
LGTM, left feedback, PTAL

**Comment by janhavisinghh** [comment]
> Steps to reproduce: New medicine -> type custom med -> click on the dialog -> as soon as start typing we see the mg.

Yes this is the expected behaviour @Honey

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**